ui-v2/app/utils/routing/walk.js

```diff
@@ -0,0 +1,80 @@
+export const walk = function(routes) {
+  const keys = Object.keys(routes);
+  keys.forEach((item, i) => {
+    if (item === '_options') {
+      return;
+    }
+    const options = routes[item]._options;
+    let cb;
+    if (Object.keys(routes[item]).length > 1) {
+      cb = function() {
+        walk.apply(this, [routes[item]]);
+      };
+    }
+    this.route(item, options, cb);
+  });
+  if (typeof routes.index === 'undefined') {
```

Comment:
This branch isn't necessary - Ember will create index routes for you if nest callbacks.

Reply:
I think I added this as it's needed from the other side i.e. if I try `isWildcard('kv.index')` it needs to return `false`, not throw. Come to think of it, I think it would be caught anyway. I added a try/catch to catch any other auto routes I wasn't aware of, the `application` route also caught me out whilst testing, so I added this then. I'd rather not have a try/catch but I don't know enough about ember to be sure I auto add all its auto created routes.

Just considering whether I'd remove it now I've realized that, even from the other side, any error would be caught. 

I think if anything I'd rather replicate ember's 'auto' routes in in the JSON blob rather than catch them - if anything I'd rather automatically add `application` in here also if you haven't defined it.